src/superstream_consumer.rs

```diff
@@ -0,0 +1,74 @@
+use crate::superstream::DefaultSuperStreamMetadata;
+use crate::{error::ConsumerCreateError, Consumer, Environment};
+use rabbitmq_stream_protocol::commands::subscribe::OffsetSpecification;
+use std::sync::Arc;
+//type FilterPredicate = Option<Arc<dyn Fn(&Message) -> bool + Send + Sync>>;
```

Comment:
For support filtering I'm having some issue here to convert back Option<FilterValueExtractor> back to dyn Fn(&Message) -> bool + Send + Sync> inside Send when creating a new Producers

Reply:
Solved!